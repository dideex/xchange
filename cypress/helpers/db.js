const mongoose = require('mongoose')
const mongoConfig = require('./config.json').mongodb

module.exports = mongoose.connect(mongoConfig, {useNewUrlParser: true})
