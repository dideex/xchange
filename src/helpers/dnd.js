import React from 'react'
import {DragDropContext} from 'react-dnd'
import TestBackend from 'react-dnd-test-backend'

export function wrapInTestContext(DecoratedComponent) {
  const TestStub = props => <DecoratedComponent {...props} />
  return DragDropContext(TestBackend)(TestStub)
}
