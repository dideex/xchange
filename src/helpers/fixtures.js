export const fakeCurrnecy = [
  {
    order: 1,
    _id: 1,
    icon: 'eth',
    id: 'eth',
    name: 'tether',
    label: 'tether label',
    source: 'tether source wallet'
  },
  {
    order: 2,
    _id: 2,
    icon: 'bitcoin',
    id: 'bitcoin',
    name: 'lisk',
    label: 'lisk label',
    source: 'lisk source wallet'
  },
  {
    order: 3,
    _id: 3,
    icon: 'rur',
    id: 'rur',
    name: 'litecoin',
    label: 'litecoin label',
    source: 'litecoin source wallet'
  },
]


export const fakeData = {
  id: 'Test id',
  inputValue: '1000',
  toWallet: 'Test wallet',
  currencyInputLabel: 'Test input label',
  currencyOutputLabel: 'Test output label',
  fromWallet: 'Tests fromWallet',
  outputValue: '2000',
  email: 'Tests email',
  username: 'Tests username',
  paymentStatus: 1,
  loading: false,
  wallets: {
    eth: 'eth wallet',
    bitcoint: 'bitcoint wallet',
    rur: 'rur wallet',
  },
  login: 'Test login',
  updatePaymentStatus() {},
}

export const fakeUser = {
  username: 'Fake username',
  password: 'Fake password',
  email: 'fake@email.com',
  phone: '89145678901',
  fio: 'Fake fio',
  token: 'Fake token',
}

export const fakeBitcoinAddress = '3JQSigWTCHyBLRD979JWgEtWP5YiiFwcQB '
export const fakeCreditAddress = '1234 4321 5678 8765'

export const fakeWallets = {
  eth: 'Fake eth wallet',
  bitcoin: 'Fake bitcoin wallet',
  rur: 'Fake rur wallet'
}