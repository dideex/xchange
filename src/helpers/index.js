export {fakeCurrnecy} from './fixtures'

export const delay = (fn = () => {}, timeout = 0) =>
  new Promise(res =>
    setTimeout(() => {
      fn()
      res()
    }, timeout),
  )
