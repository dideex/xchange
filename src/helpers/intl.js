import React from 'react'
import {mount, shallow} from 'enzyme'
import {IntlProvider, intlShape} from 'react-intl'

import messages from '../locale/en.json'
const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

function nodeWithIntlProp(node) {
  return React.cloneElement(node, {intl})
}

export function mountWithIntl(
  node,
  {context, childContextTypes, ...additionalOptions} = {},
) {
  return mount(nodeWithIntlProp(node), {
    context: Object.assign({}, context, {intl}),
    childContextTypes: Object.assign({}, {intl: intlShape}, childContextTypes),
    ...additionalOptions,
  })
}
