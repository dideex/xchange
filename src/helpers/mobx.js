import React from 'react'
import {Provider} from 'mobx-react'
import {lastOperations, cash, user} from '../store'

jest.mock('../components/Api', () => ({
  post: () => Promise.resolve({data: {}}),
  get: () => Promise.resolve({data: {}}),
  errorEmitter: jest.fn(data => fn => data(fn)),
}))

// jest.mock('../components/common', () => ({
//   noty: () => {},
//   Loading: () => <div>Loading</div>,
// }))

export const MobxProvider = ({children, ...stores}) => {
  return (
    <Provider
      lastOperationsStore={lastOperations}
      cashStore={cash}
      userStore={user}
      {...stores}
    >
      {children}
    </Provider>
  )
}
