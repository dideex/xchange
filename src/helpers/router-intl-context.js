import {mount} from 'enzyme'
import {BrowserRouter} from 'react-router-dom'
import {IntlProvider, intlShape} from 'react-intl'
import {shape} from 'prop-types'

import messages from '../locale/en.json'

export const testId = 'test id'

// Instantiate router context
const router = {
  history: {...new BrowserRouter().history},
  route: {
    location: {
      path: '/location',
      href: '/location',
    },
    match: {
      path: '/location',
      params: {
        id: testId,
      },
    },
  },
}

// const createContext = () => ({
//   context: {router},
//   childContextTypes: {router: shape({})},
// })

const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

const createMixedContext = ({
  router,
  context,
  childContextTypes,
  ...additionalOptions
} = {}) => ({
  context: Object.assign({}, {router}, context, {intl}),
  childContextTypes: Object.assign(
    {},
    {router: shape({})},
    {intl: intlShape},
    childContextTypes,
  ),
  ...additionalOptions,
})

export function mountWrap(node) {
  return mount(node, createMixedContext({router}))
}

// export function shallowWrap(node) {
//   return shallow(node, createContext())
// }
