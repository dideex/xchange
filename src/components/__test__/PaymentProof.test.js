import React from 'react'
import {mount} from 'enzyme'
import Component, {PaymentProof} from '../PaymentProof'
import {Provider as MobxProvider} from 'mobx-react'

import {mountWrap} from '../../helpers/router-intl-context'
import {fakeCurrnecy, fakeData, fakeWallets, fakeUser} from '../../helpers/fixtures'
import UserStore from '../../store/User'
import CashStore from '../../store/Cash'

jest.mock('../Api', () => ({
  post: () => Promise.resolve({data: {}}),
  get: () => Promise.resolve({data: {}, eth: {}}),
  errorEmitter: fn => data => fn(data),
}))

jest.mock('../common/Noty.js', () => ({
  noty: () => {},
}))

describe('Paymentproof', () => {
  let userStore = new UserStore()
  let cashStore = new CashStore()
  beforeEach(() => {
    userStore = new UserStore()
    cashStore = new CashStore()
  })

  describe('Markup', () => {
    it('Basic markup', () => {
      cashStore.currency = fakeCurrnecy
      cashStore.orderId = fakeData.id
      cashStore.inputValue = fakeData.inputValue
      cashStore.outputValue = fakeData.outputValue
      userStore.email = fakeData.email
      userStore.username = fakeData.username
      userStore.wallets = fakeWallets
      const wrapper = mountWrap(
        <MobxProvider {...{userStore, cashStore}}>
          <Component />
        </MobxProvider>,
      )

      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  describe('Component behavriour', () => {
    it('cDM should scroll the window', () => {
      cashStore.currency = fakeCurrnecy
      cashStore.orderId = fakeData.id
      cashStore.inputValue = fakeData.inputValue
      cashStore.outputValue = fakeData.outputValue
      userStore.email = fakeData.email
      userStore.username = fakeData.username
      userStore.wallets = fakeWallets

      window.scrollTo = jest.fn()
      const wrapper = mountWrap(
        <MobxProvider {...{userStore, cashStore}}>
          <Component />
        </MobxProvider>,
      )

      expect(window.scrollTo).toHaveBeenCalledTimes(1)
      expect(window.scrollTo).toHaveBeenCalledWith(0, -150)
    })

    describe('Router behaviour', () => {
      it('Component should redirect to the home when payment status is 0', () => {
        cashStore.currency = fakeCurrnecy
        cashStore.paymentStatus = 0
        userStore.wallets = fakeWallets
        const formatMessage = () => {}
        const push = jest.fn()

        const wrapper = mount(
          <PaymentProof
            {...{userStore, cashStore}}
            intl={{formatMessage}}
            history={{push}}
          />,
        )

        expect(push).toHaveBeenCalledTimes(1)
        expect(push).toHaveBeenCalledWith('/')
      })

      it("Component should redirect to the 'thank you' page when payment status is 2", () => {
        cashStore.currency = fakeCurrnecy
        cashStore.paymentStatus = 2
        userStore.wallets = fakeWallets
        const formatMessage = () => {}
        const push = jest.fn()

        const wrapper = mount(
          <PaymentProof
            {...{userStore, cashStore}}
            intl={{formatMessage}}
            history={{push}}
          />,
        )

        expect(push).toHaveBeenCalledTimes(1)
        expect(push).toHaveBeenCalledWith('/spasibo')
      })

      it('The go back button should work correclty', () => {
        cashStore.currency = fakeCurrnecy
        userStore.wallets = fakeWallets
        const formatMessage = () => {}
        const push = () => {}
        const goBack = jest.fn()

        const wrapper = mount(
          <PaymentProof
            {...{userStore, cashStore}}
            intl={{formatMessage}}
            history={{push, goBack}}
          />,
        )
        wrapper.find('[data-testid="payment-proof-back"]').simulate('click')

        expect(goBack).toHaveBeenCalledTimes(1)
      })

      it('The confirm payment button should work correclty', () => {
        cashStore.currency = fakeCurrnecy
        cashStore.paymentStatus = 1
        cashStore.cofirmPayment = jest.fn()
        userStore.wallets = fakeWallets
        userStore.email = fakeUser.email
        const formatMessage = () => {}
        const push = jest.fn()

        const wrapper = mount(
          <PaymentProof
            {...{userStore, cashStore}}
            intl={{formatMessage}}
            history={{push}}
          />,
        )
        wrapper.find('[data-testid="confirm-payment"]').simulate('click')

        expect(push).toHaveBeenCalledTimes(1)
        expect(push).toHaveBeenCalledWith('/spasibo')
        expect(cashStore.cofirmPayment).toHaveBeenCalledTimes(1)
        expect(cashStore.cofirmPayment).toHaveBeenCalledWith(fakeUser.email)
      })
    })
  })
})
