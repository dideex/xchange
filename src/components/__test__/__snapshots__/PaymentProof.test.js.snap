// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paymentproof Markup Basic markup 1`] = `"<div class=\\"css-11c6jtf\\"><div><h2 class=\\"css-x184e\\"><span>Make a transfer</span></h2></div><div class=\\"css-1vwsum4\\"><h3>Перевод № </h3><div class=\\"css-ubpv4x\\"><div><div class=\\"css-ubpv4x\\"><span><span>Wallet for transfer:</span></span><strong></strong></div><div class=\\"css-ubpv4x\\"><span><span>Amount to transfer:</span></span><strong>0 undefined</strong></div></div><div><div class=\\"css-ubpv4x\\"><span><span>Get on wallet:</span></span><strong></strong></div><div class=\\"css-ubpv4x\\"><span><span>Amount received:</span></span><strong>0 undefined</strong></div></div></div><div class=\\"css-f37lyo\\"><p><span>Email:</span><strong></strong></p><p><span><span>Full name:</span></span><strong></strong></p><p><span>Status:</span><strong><span>Не создан</span></strong></p></div></div><p><span role=\\"img\\" aria-label=\\"back\\">👈</span><span>Back</span></p><button cursor=\\"pointer\\" class=\\"css-8hkfp7\\"><div class=\\"css-1v1d0dg\\">I translated</div></button><svg class=\\"howto__step2\\" viewBox=\\"0 0 298 373\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M177.988 333.829H4.88133C2.16948 333.829 0 331.658 0 328.946V221.338C0 218.625 2.16948 216.455 4.88133 216.455H177.988C180.699 216.455 182.869 218.625 182.869 221.338V328.946C182.779 331.658 180.609 333.829 177.988 333.829Z\\" fill=\\"#BBCADB\\"></path><path d=\\"M79.186 333.829C80.0899 344.861 73.5815 353.361 63.2765 360.595H115.344C105.943 353.813 100.881 344.861 100.158 333.829H79.186Z\\" fill=\\"#BBCADB\\"></path><path d=\\"M172.564 227.035H10.757V323.791H172.564V227.035Z\\" fill=\\"white\\"></path><path d=\\"M22.0564 323.882V283.28C22.0564 282.105 23.0507 281.11 24.2259 281.11H156.383C157.558 281.11 158.553 282.105 158.553 283.28V323.882H22.0564Z\\" fill=\\"#A5C957\\"></path><path d=\\"M40.5874 272.519C50.8217 272.519 59.1183 264.22 59.1183 253.982C59.1183 243.744 50.8217 235.445 40.5874 235.445C30.353 235.445 22.0564 243.744 22.0564 253.982C22.0564 264.22 30.353 272.519 40.5874 272.519Z\\" fill=\\"#8EB85B\\"></path><path d=\\"M160.18 243.673H69.8753C69.6945 243.673 69.4233 243.492 69.4233 243.221V238.157C69.4233 237.976 69.6041 237.705 69.8753 237.705H160.18C160.361 237.705 160.632 237.886 160.632 238.157V243.221C160.632 243.492 160.451 243.673 160.18 243.673Z\\" fill=\\"#DEE0E0\\"></path><path d=\\"M77.5588 253.982H69.8753C69.6945 253.982 69.4233 253.801 69.4233 253.53V248.466C69.4233 248.285 69.6041 248.014 69.8753 248.014H77.5588C77.7396 248.014 78.0108 248.195 78.0108 248.466V253.53C77.9204 253.801 77.7396 253.982 77.5588 253.982Z\\" fill=\\"#DEE0E0\\"></path><path d=\\"M138.756 253.982H80.9938C80.8131 253.982 80.5419 253.801 80.5419 253.53V248.466C80.5419 248.285 80.7227 248.014 80.9938 248.014H138.756C138.937 248.014 139.208 248.195 139.208 248.466V253.53C139.118 253.801 138.937 253.982 138.756 253.982Z\\" fill=\\"#DEE0E0\\"></path><path d=\\"M127.457 265.285H69.8753C69.6945 265.285 69.4233 265.104 69.4233 264.833V259.769C69.4233 259.588 69.6041 259.317 69.8753 259.317H127.457C127.638 259.317 127.909 259.498 127.909 259.769V264.833C127.909 265.104 127.728 265.285 127.457 265.285Z\\" fill=\\"#DEE0E0\\"></path><path d=\\"M87.7735 297.206H30.5535C30.1919 297.206 29.9207 296.935 29.9207 296.573V291.961C29.9207 291.599 30.1919 291.328 30.5535 291.328H87.7735C88.1351 291.328 88.4062 291.599 88.4062 291.961V296.573C88.4062 296.935 88.1351 297.206 87.7735 297.206Z\\" fill=\\"white\\"></path><path d=\\"M87.7735 309.413H30.5535C30.1919 309.413 29.9207 309.142 29.9207 308.78V304.169C29.9207 303.807 30.1919 303.536 30.5535 303.536H87.7735C88.1351 303.536 88.4062 303.807 88.4062 304.169V308.78C88.4062 309.142 88.1351 309.413 87.7735 309.413Z\\" fill=\\"white\\"></path><path d=\\"M87.7735 321.621H30.5535C30.1919 321.621 29.9207 321.35 29.9207 320.988V316.376C29.9207 316.014 30.1919 315.743 30.5535 315.743H87.7735C88.1351 315.743 88.4062 316.014 88.4062 316.376V320.988C88.4062 321.35 88.1351 321.621 87.7735 321.621Z\\" fill=\\"white\\"></path><path d=\\"M212.077 165.159C242.472 134.754 242.472 85.4577 212.077 55.0529C181.683 24.6481 132.404 24.6481 102.01 55.0529C71.6159 85.4577 71.6159 134.754 102.01 165.159C132.404 195.563 181.683 195.563 212.077 165.159Z\\" fill=\\"#EDDA48\\"></path><path d=\\"M203.448 156.527C229.077 130.889 229.077 89.3224 203.448 63.6849C177.82 38.0474 136.268 38.0474 110.639 63.6849C85.0105 89.3224 85.0105 130.889 110.639 156.527C136.268 182.164 177.82 182.164 203.448 156.527Z\\" fill=\\"#E2B255\\"></path><path d=\\"M176.18 125.667C199.863 127.928 196.609 104.869 186.214 95.2837L196.699 84.7942L189.558 77.6505L178.982 88.2304L175.276 84.5229L185.852 73.943L179.163 67.2515L168.677 77.741L156.293 65.3525L150.056 71.592L157.92 79.4591L124.835 112.555L116.971 104.688L110.643 111.018L123.028 123.406L112.723 133.715L119.231 140.226L129.717 129.736L133.604 133.624L123.118 144.114L129.717 150.715L140.203 140.226C155.841 155.056 179.524 153.699 176.18 125.667ZM168.406 91.2145L177.897 100.709C181.151 104.778 177.445 115.63 166.327 112.284L156.745 102.88L168.406 91.2145ZM147.163 134.077L136.406 123.316L150.056 109.661L160.813 120.422C166.056 130.369 158.824 138.779 147.163 134.077Z\\" fill=\\"#EDDA48\\"></path><path d=\\"M157.016 357.249C157.016 357.249 160.632 372.802 157.016 372.893C153.4 372.983 145.988 373.526 136.858 370.903C119.592 367.286 119.592 368.01 119.592 362.403C119.592 356.887 130.711 359.329 134.508 354.627C138.304 349.924 157.016 357.249 157.016 357.249Z\\" fill=\\"#35291F\\"></path><path d=\\"M248.767 350.738C248.767 350.738 237.377 351.733 237.377 355.621C235.388 365.749 245.241 364.755 250.394 366.02C256.27 366.653 260.518 366.02 260.518 366.02C262.778 369.276 269.015 371.898 273.897 369.276C278.778 366.653 271.004 348.749 271.004 348.749L248.767 350.738Z\\" fill=\\"#35291F\\"></path><path d=\\"M171.75 201.444C160.541 211.03 150.96 232.008 142.01 258.323L132.248 352.095C139.027 361.409 147.344 363.127 157.016 357.339C160.903 323.52 169.31 289.158 166.146 273.333C178.259 255.067 190.643 236.349 203.208 217.359C203.208 217.359 210.259 264.11 212.338 268.812C214.417 273.514 248.767 350.829 248.767 350.829C257.174 354.446 264.496 353.542 270.914 348.839C258.53 315.201 253.287 281.924 238.371 269.445L243.705 210.216L171.75 201.444Z\\" fill=\\"#565F88\\"></path><path d=\\"M150.959 179.742C163.253 163.375 175.637 146.646 191.728 132.901L214.326 79.9112C219.84 74.9378 226.801 74.0335 234.304 74.6665L208.722 56.5812L218.756 44.2832L291.614 65.0813C303.908 78.1931 297.851 101.794 278.597 128.018L287.275 188.333C258.71 216.998 217.852 215.551 171.841 201.354L173.558 180.556L159.637 191.859L150.959 179.742Z\\" fill=\\"#394B5F\\"></path><path d=\\"M224.451 23.5754L221.468 28.7297C221.106 37.2298 222.824 44.5544 227.705 50.3417L261.241 58.6609L259.885 51.3364L224.451 23.5754Z\\" fill=\\"#E7D6BA\\"></path><path d=\\"M261.241 58.661L251.5 55L241.354 53.6875C241.354 57.2141 243.614 60.2886 241.354 66.8898L261.241 58.661Z\\" fill=\\"#E7D6BA\\"></path><path d=\\"M205.197 19.6872C205.197 19.6872 212.97 21.6766 215.954 19.6872C218.937 17.6978 226.801 8.92643 236.744 6.66577C256.36 8.92643 261.965 18.0595 267.027 27.4639C268.925 32.8894 264.405 42.1129 259.885 51.246C254.371 53.4162 249.4 49.7088 244.609 44.2832C241.083 40.0331 236.112 38.6767 229.332 41.4799C230.055 35.3309 228.518 31.0809 224.451 28.9107C224.451 28.9107 215.954 28.4585 211.705 26.8309C207.456 25.2032 205.197 19.6872 205.197 19.6872Z\\" fill=\\"#394B5F\\"></path><path d=\\"M217.4 45.9107L212.518 51.9693H208.722C200.406 44.5543 190.914 39.3096 180.88 36.0542C186.304 35.8734 200.677 34.1553 205.196 36.1446C209.716 38.134 217.4 45.9107 217.4 45.9107Z\\" fill=\\"#E7D6BA\\"></path><path d=\\"M157.92 189.418C157.92 189.418 152.315 197.104 145.807 197.104C139.299 197.104 134.508 193.125 133.423 189.418C132.338 185.71 134.96 184.625 134.96 184.625L152.858 182.274L157.92 189.418Z\\" fill=\\"#E7D6BA\\"></path></svg></div>"`;
